packages/apps/fortune/exchange-oracle/server/src/common/guards/signature.auth.ts

```diff
@@ -20,6 +21,9 @@ export class SignatureAuthGuard implements CanActivate {
     const signature = request.headers[HEADER_SIGNATURE_KEY];
     const oracleAdresses: string[] = [];
     try {
+      if (this.role.includes(Role.Worker)) {
+        return verifySignature(data, signature, [data.senderAddress]);
```

Comment:
The DTO you're using in this endpoint doesn't contain the sender address. You should modify the DTO of the endpoint by requesting now the assignmentId. With that, you can check the workerAddress in the database for verifying and also you can get escrowAddress and chainId from there.